src/core_scan/Core_scan.ml

```diff
@@ -477,12 +477,25 @@ let parse_equivalences equivalences_file =
 (* Iteration helpers *)
 (*****************************************************************************)
 
-let iter_targets_and_get_matches_and_exn_to_errors config f targets =
+(*
+   handle_target returns (results, was_scanned) where was_scanned indicates
+   whether at least one rule applied to the target since the target could
+   be excluded by all the rules via per-rule include/exclude patterns.
+   (sorry about the complexity; baking this flag into match_result type
+   would lead to even worse complexity)
+
+   Returns a list of match results and a separate list of scanned targets.
+*)
+let iter_targets_and_get_matches_and_exn_to_errors config
+    (handle_target :
+      In.target -> Core_profiling.partial_profiling RP.match_result * bool)
```

Comment:
Could it be
```ocaml
[ `Scanned of Core_profiling.partial_profiling RP.match_result
| `NotScanned
]
```
?

Reply:
Oh actually this is better than the suggestion I made to use a record

Reply:
`match_result` contains an `errors` field and an `extra` field for profiling info so I think they have to be returned even if no rule was used sucessfully.

Reply:
My understanding was that "not scanned" here means that the target wasn't selected by any rule, therefore nothing run, so there should be no errors nor any profiling data to report. Looking at the `scan` function seems to corroborate this too, since `was_scanned` is `false` if `applicable_rules` is empty. But if you still think that a `match_result` is needed for "not scanned" targets then I trust you.